### PR TITLE
Add support for builtin sum

### LINKED ISSUE #18627: [jit] Python built-in function support
We support many built-in functions ([Python 2](https://docs.python.org/2/library/functions.html), [Python 3](https://docs.python.org/3/library/functions.html)), but not all of them. Some of these need some more discussion (i.e. what about builtins that exist in Python 2 but not 3 and vice versa), but many can be done piecemeal and should be straightforward to implement. When you implement a builtin please check to see if it is a [magic method](https://rszalski.github.io/magicmethods/).

To do
- [x] `abs()`
- [ ] `all()`
- [ ] `any()`
- [x] `bin()`
- [x] `bool()`
- [ ] `bytearray()`
- [x] `chr()`
- [x] `dict()`
- [x] `divmod()`
- [x] `enumerate()`
- [x] `float()`
- [x] `getattr()`
- [x] `hash()`
- [x] `hex()`
- [x] `int()`
- [x] `isinstance()`
- [x] `len()`
- [x] `list()`
- [x] `long()` (long is deprecated in py3 builtins)
- [x] `max()`
- [x] `min()`
- [x] `oct()`
- [x] `ord()`
- [x] `pow()`
- [x] `print()`
- [x] `range()`
- [x] `reversed()`
- [x] `round()`
- [ ] `set()`
- [ ] `slice()`
- [x] `sorted()` ([separate issue](https://github.com/pytorch/pytorch/issues/19372), implemented for lists)
- [x] `str()`
- [ ] `sum()`
- [ ] `type()`

Maybe to do later (when things like first class modules are in, or if we ever have something like iterators)
- [ ] `callable()`
- [x] `tuple()`
- [ ] `complex()`
- [ ] `delattr()`
- [ ] `filter()`
- [ ] `format()`
- [ ] `frozenset()`
- [ ] `globals()`
- [ ] `hasattr()`
- [ ] `issubclass()`
- [ ] `iter()`
- [ ] `locals()`
- [ ] `next()`
- [ ] `object()`
- [ ] `property()`
- [ ] `reduce()`
- [ ] `setattr()`
- [ ] `repr()`
- [ ] `super()`
- [ ] `vars()`
- [x] `zip()`

Probably won't do
- [ ] `__import__()`
- [ ] `basestring()`
- [ ] `classmethod()`
- [ ] `cmp()`
- [ ] `compile()`
- [ ] `dir()`
- [ ] `eval()`
- [ ] `execfile()`
- [ ] `file()`
- [ ] `help()`
- [ ] `id()`
- [ ] `input()`
- [ ] `map()`
- [ ] `memoryview()`
- [ ] `open()`
- [ ] `raw_input()`
- [ ] `reload()`
- [ ] `staticmethod()`
- [ ] `unichr()`
- [ ] `unicode()`
- [ ] `xrange()`

cc @suo

### PR DESCRIPTION
**Pull request by nikithamalgifb**
Fixes #18627

Summary:
Adds torch.sum support for JIT

Test:
python test/test_jit.py -k test_list_sum
python test/test_jit.py -k test_torch_sum

### CODE REVIEW

File: torch/csrc/jit/runtime/register_prim_ops_fulljit.cpp [hunk 1]
```diff
@@ -1007,6 +1007,49 @@ RegisterOperators reg2({
...
+    Operator(
+        "aten::sum.float(float[] self) -> float",
+        [](Stack* stack) {
+          c10::List<double> l = pop(stack).toDoubleList();
+          auto sum = 0.0;
+          for (const auto& elem : l) {
+            sum += elem;
+          }
+
+          // To handle cases where the List only contains int,
```
Comment (gmagogsfm) [inline torch/csrc/jit/runtime/register_prim_ops_fulljit.cpp:1030]: OOC, why is this special case needed?

Comment (nikithamalgifb) [reply torch/csrc/jit/runtime/register_prim_ops_fulljit.cpp:1030]: This is to handle cases where the the List has been annotated as `float` but it only contains `integer` elements. Then the sum of all the elements of the list will also be `int`. So this case handles that. Here is the example. 

```
def (x : List[float]):
       return sum(x)
x = [1, 2, 3]
```

Comment (eellison) [reply torch/csrc/jit/runtime/register_prim_ops_fulljit.cpp:1030]: if we have a int list inside a float[] operator i think that's a bug that should be fixed elsewhere.

Comment (nikithamalgifb) [reply torch/csrc/jit/runtime/register_prim_ops_fulljit.cpp:1030]: @eellison : So ideally, we should make the float to a int list then?

Comment (nikithamalgifb) [reply torch/csrc/jit/runtime/register_prim_ops_fulljit.cpp:1033]: Ok, confirmed with @eellison offline and we should not support int list inside float[]. I have removed this part of the code snippet in the new version. Please review.

File: test/test_jit.py [hunk 2]
```diff
@@ -1045,6 +1045,35 @@ def fn(*args):
                 self.assertFalse(fn.has_trace_for(*unk_config))
         self.assertEqual(fn.hits, 0)
 
+    def test_torch_sum(self):
```
Comment (gmagogsfm) [inline test/test_jit.py:1048]: Is `torch.sum` supported added in this PR?

File: torch/csrc/jit/runtime/register_prim_ops_fulljit.cpp [hunk 3]
```diff
@@ -1007,6 +1007,41 @@ RegisterOperators reg2({
           push(stack, std::move(t));
         },
         aliasAnalysisFromSchema()),
+    Operator(
```
[the remark this answers is not in the document]
Comment (nikithamalgifb) [reply torch/csrc/jit/runtime/register_prim_ops_fulljit.cpp:1010]: Ok, all the builtin  ops are registered in this file `register_prim_ops_fulljit.cpp`.

File: test/test_jit.py (hunk 2 above)
Comment (nikithamalgifb) [reply test/test_jit.py:1048]: Yes, torch.sum is supported in this PR.

**Review by gmagogsfm — APPROVED** [review]

**Pull request closed without merging**